File: sdkwork-ai-sdk-typescript/src/api/music.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { MusicGenerationRequest, SunoMusic, SunoMusicDeleteResponse, SunoMusicList } from '../types';


export class MusicApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Generate music */
  async createGenerate(body: MusicGenerationRequest): Promise<SunoMusic> {
    return this.client.post<SunoMusic>(aiApiPath(`/v1/music/generations`), body);
  }

/** List music */
  async listMusic(params?: QueryParams): Promise<SunoMusicList> {
    return this.client.get<SunoMusicList>(aiApiPath(`/music`), params);
  }

/** Generate music */
  async createGenerateMusic(body: MusicGenerationRequest): Promise<SunoMusic> {
    return this.client.post<SunoMusic>(aiApiPath(`/music`), body);
  }

/** Generate music */
  async createGenerateGenerations(body: MusicGenerationRequest): Promise<SunoMusic> {
    return this.client.post<SunoMusic>(aiApiPath(`/music/generations`), body);
  }

/** Retrieve music */
  async retrieve(music_id: string | number): Promise<SunoMusic> {
    return this.client.get<SunoMusic>(aiApiPath(`/music/${music_id}`));
  }

/** Delete music */
  async deleteMusic(music_id: string | number): Promise<SunoMusicDeleteResponse> {
    return this.client.delete<SunoMusicDeleteResponse>(aiApiPath(`/music/${music_id}`));
  }
}

export function createMusicApi(client: HttpClient): MusicApi {
  return new MusicApi(client);
}

File: sdkwork-ai-sdk-go/api/music.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type MusicApi struct {
    client *sdkhttp.Client
}

func NewMusicApi(client *sdkhttp.Client) *MusicApi {
    return &MusicApi{client: client}
}

// Generate music
func (a *MusicApi) CreateGenerate(body sdktypes.MusicGenerationRequest) (sdktypes.SunoMusic, error) {
    raw, err := a.client.Post(AiApiPath("/v1/music/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.SunoMusic
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusic](raw)
}

// List music
func (a *MusicApi) ListMusic(query map[string]interface{}) (sdktypes.SunoMusicList, error) {
    raw, err := a.client.Get(AiApiPath("/music"), query, nil)
    if err != nil {
        var zero sdktypes.SunoMusicList
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusicList](raw)
}

// Generate music
func (a *MusicApi) CreateGenerateMusic(body sdktypes.MusicGenerationRequest) (sdktypes.SunoMusic, error) {
    raw, err := a.client.Post(AiApiPath("/music"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.SunoMusic
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusic](raw)
}

// Generate music
func (a *MusicApi) CreateGenerateGenerations(body sdktypes.MusicGenerationRequest) (sdktypes.SunoMusic, error) {
    raw, err := a.client.Post(AiApiPath("/music/generations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.SunoMusic
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusic](raw)
}

// Retrieve music
func (a *MusicApi) Retrieve(music_id string) (sdktypes.SunoMusic, error) {
    raw, err := a.client.Get(AiApiPath(fmt.Sprintf("/music/%s", music_id)), nil, nil)
    if err != nil {
        var zero sdktypes.SunoMusic
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusic](raw)
}

// Delete music
func (a *MusicApi) DeleteMusic(music_id string) (sdktypes.SunoMusicDeleteResponse, error) {
    raw, err := a.client.Delete(AiApiPath(fmt.Sprintf("/music/%s", music_id)), nil, nil)
    if err != nil {
        var zero sdktypes.SunoMusicDeleteResponse
        return zero, err
    }
    return decodeResult[sdktypes.SunoMusicDeleteResponse](raw)
}
